contracts/interchain-token-service/src/state.rs

```diff
@@ -126,10 +126,19 @@ pub struct TokenConfig {
     pub origin_chain: ChainNameRaw,
 }
 
+type TokenAddress = nonempty::HexBinary;
+#[cw_serde]
+pub struct CustomToken {
+    pub chain: ChainNameRaw,
```

Comment:
Why is the chain included here?

Reply:
it's not necessary, but we do this in some maps and not others (include the key in the object itself). Not sure what our pattern should be

Reply:
I'm still of the opinion that we shouldn't need to rely on the storage key to get relevant data. Keys should be pretty much one way, i.e. you can compose them from intelligible data and you can use them to filter directly in the store, but they shouldn't be treated as part of the value. Otherwise, we tie the storage model to the contract logic and it's a pain to make changes. I'm fine with normalizing data in the store, i.e. we could fill a state object with multiple calls to different keys in the storage.

Reply:
This pattern feels pretty weird when we have structured keys that are part of the storage themselves. I understand that this was problematic in core, and can be an issue in other VMs like EVM where the key isn't stored as well, but cosmwasm guarantees both key and value can be retrieved from storage so this feels unnecessary. It introduces a new concern where the chain name / token address in the key don't match what's stored in the value which I don't really like. I'm fine with sticking to this if you strongly favour it, but I don't quite see the benefit of doing this everywhere

Reply:
I do prefer just storing all the relevant data together in one object